File: source-connector/src/main/java/sourceconnector/domain/log/factory/LogFactory.java
package sourceconnector.domain.log.factory;

import sourceconnector.domain.log.Log;
import sourceconnector.domain.log.LogMetadata;

public interface LogFactory {
  /**
   * @param payload log payload
   * @param logMetadata which is origin
   * @return Log
   */
  Log create(String payload, LogMetadata logMetadata);
}

File: source-connector/src/main/kotlin/sourceconnector/domain/log/factory/LogFactory.kt
package sourceconnector.domain.log.factory

import sourceconnector.domain.log.Log
import sourceconnector.domain.log.LogMetadata

interface LogFactory {
  /**
   * @param payload log payload
   * @param logMetadata which is origin
   * @return Log
   */
  fun create(payload: String, logMetadata: LogMetadata): Log
}
